### PR TITLE
pica: use correct coordinates for texture 2

### PR DESCRIPTION
**Pull request by wwylele**
See [here](https://www.3dbrew.org/wiki/GPU/Internal_Registers#GPUREG_TEXUNIT_CONFIG) for documentation of the register in question. We currently only respect the low 3 bits. Texture 2 coordinates should be chosen according to bit 13.

[Here is a hardware test](https://gist.github.com/wwylele/fb8306a1facbff5fe77f5db7142f78ef) adapted from the "graphics/gpu/textured_cube" example program. In the test, the texture is bind as texture 2, and the shader outputs the original texture coordinates to coord2, and the negated coordinates to coord1. Press A to flip bit 13 of the register, and texture flipping, as the result of switching coordinates, can be observed.

Don't know if this fix any games.

### CODE REVIEW

**Comment by wwylele** [comment]
btw, we should take some time to consolidate our pica register layout according to 3dbrew documentation. Would be great if we can have some easy way to detect if a game is accessing an unimplemented register

File: src/video_core/regs_texturing.h [hunk 1]
```diff
@@ -122,6 +122,10 @@ struct TexturingRegs {
         BitField<0, 1, u32> texture0_enable;
         BitField<1, 1, u32> texture1_enable;
         BitField<2, 1, u32> texture2_enable;
+        BitField<8, 2, u32> texture3_coordinates;
+        BitField<10, 1, u32> texture3_enable;
+        BitField<13, 1, u32> texture2_use_coord1;
+        BitField<16, 1, u32> clear_texture_cache;
```
Comment (JayFoxRox) [inline src/video_core/regs_texturing.h]: This is dangerous: adding unused registers will probably cause lot of confusion when finding remaining broken features

File: src/video_core/regs_texturing.h [hunk 2]
```diff
@@ -122,6 +122,10 @@ struct TexturingRegs {
         BitField<0, 1, u32> texture0_enable;
         BitField<1, 1, u32> texture1_enable;
         BitField<2, 1, u32> texture2_enable;
+        BitField<8, 2, u32> texture3_coordinates;
```
Comment (JayFoxRox) [inline src/video_core/regs_texturing.h]: Please also add support for this while you are at it

**Comment by JayFoxRox** [comment]
This feature is necessary for Kid Icarus. I've a blog article lined up for this.

My implementation from December (rebasing it right now) can be found at https://github.com/JayFoxRox/citra/pull/47

**Comment by wwylele** [comment]
Good point, I am trying to figure out the bump mapping stuff. My first guess is yes.
Just want to make sure: there is not other place using textures, right?

**Comment by wwylele** [comment]
Fixed bump map selector in hw renderer (sw one doesn't implement fragment lighting). Still left texcoord3 unimplemented because procedural texture sampling might go through a complete different routine in the future implementation.

**Comment by wwylele** [comment]
hold on, I forgot to make `shader_dirty = true`

**Comment by wwylele** [comment]
Update: added `shader_dirty=true` for texture2 coordinates change

**Pull request merged**